### DISCUSSION TITLE
Patchable metadata for beatmaps with online association

### DISCUSSION THREAD

**Discussion by huoyaoyuan** [ppy/osu#13080] [Ideas]
**Describe the new feature:** 
Beatmap metadata may be updated after ranked:
- A new widely-fit field is introduced. For example, every old beatmap has only ACSII artist and title before 2012(or 2013?). There should be some mechanism to update them with Unicode fields.
- The creator has changed their user name, and want to reflect the change in all maps.

**Proposal designs of the feature:** 
Previously, this may require a re-download.
Proposed changes: Online APIs of beatmaps may return a new field `MetadataTimestamp`. The client checks it with stored one. If there's a mismatch and an option of automatic update is enabled, then the clients queries latest metadata and overrides with them.

The metadata must not be contained in the hash of permanent representation. Or, add new `OverrideXXX` fields besides the original ones, and then use them for priority.

**Comment by peppy** [comment]
metadata storage is planned to heavily change, including the ability to update all existing beatmaps server-side and also storing metadata against a set, not always per-difficulty.
